src/pages/workspace/AccessOrNotFoundWrapper.tsx

```diff
@@ -0,0 +1,121 @@
+
+    /** Indicated whether the report data is loading */
+    isLoadingReportData: OnyxEntry<boolean>;
+};
+
+type AccessOrNotFoundWrapperProps = AccessOrNotFoundWrapperOnyxProps & {
+    /** The children to render */
+    children: ((props: AccessOrNotFoundWrapperOnyxProps) => React.ReactNode) | React.ReactNode;
+
+    /** The report currently being looked at */
+    policyID: string;
+
+    /** Defines which types of access should be verified */
+    accessVariants?: PolicyAccessVariant[];
+
+    /** The current feature name that the user tries to get access to */
+    featureName?: PolicyFeatureName;
+
+    /** Props for customizing fallback pages */
+    fullPageNotFoundViewProps?: FullPageNotFoundViewProps;
+} & Pick<FullPageNotFoundViewProps, 'subtitleKey' | 'onLinkPress'>;
+
+type PageNotFoundFallbackProps = Pick<AccessOrNotFoundWrapperProps, 'policyID' | 'fullPageNotFoundViewProps'> & {shouldShowFullScreenFallback: boolean};
+
+function PageNotFoundFallback({policyID, shouldShowFullScreenFallback, fullPageNotFoundViewProps}: PageNotFoundFallbackProps) {
+    return shouldShowFullScreenFallback ? (
+        <FullPageNotFoundView
+            shouldShow
+            onBackButtonPress={() => Navigation.goBack(ROUTES.SETTINGS_WORKSPACES)}
+            shouldForceFullScreen
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...fullPageNotFoundViewProps}
+        />
+    ) : (
+        <NotFoundPage
+            onBackButtonPress={() => Navigation.goBack(ROUTES.WORKSPACE_PROFILE.getRoute(policyID))}
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...fullPageNotFoundViewProps}
+        />
+    );
+}
+
+function AccessOrNotFoundWrapper({accessVariants = [], fullPageNotFoundViewProps, ...props}: AccessOrNotFoundWrapperProps) {
+    const {policy, policyID, featureName, isLoadingReportData} = props;
+
+    const isPolicyIDInRoute = !!policyID?.length;
+
+    useEffect(() => {
+        if (!isPolicyIDInRoute || !isEmptyObject(policy)) {
+            // If the workspace is not required or is already loaded, we don't need to call the API
+            return;
+        }
+
+        Policy.openWorkspace(policyID, []);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isPolicyIDInRoute, policyID]);
+
+    const shouldShowFullScreenLoadingIndicator = isLoadingReportData !== false && (!Object.entries(policy ?? {}).length || !policy?.id);
+
+    const isFeatureEnabled = featureName ? PolicyUtils.isPolicyFeatureEnabled(policy, featureName) : true;
+
+    const isPageAccessible = accessVariants.reduce((acc, variant) => {
+        const accessFunction = POLICY_ACCESS_VARIANTS[variant];
+        return acc && accessFunction(policy);
+    }, true);
+
+    const shouldShowNotFoundPage = isEmptyObject(policy) || (Object.keys(policy).length === 1 && !isEmptyObject(policy.errors)) || !policy?.id || !isPageAccessible || !isFeatureEnabled;
```

Comment:
Coming from https://github.com/Expensify/App/issues/44206:
Please be careful when fix conflict.
Recent code in `FeatureEnabledAccessOrWrapper` was not applied to this file so [the fix](https://github.com/Expensify/App/pull/40650) of https://github.com/Expensify/App/issues/39443 was partially reverted.
More details: https://github.com/Expensify/App/issues/44206#issuecomment-2202252254